zmq/sugar/context.py

```diff
@@ -46,10 +43,16 @@ def __init__(self, io_threads=1, **kwargs):
         else:
             self._shadow = False
         self.sockopts = {}
-        self._sockets = set()
+        self._sockets = WeakSet()
 
     def __del__(self):
         """deleting a Context should terminate it, without trying non-threadsafe destroy"""
+
+        # Referring sys module here conceals issue #1167 on at least CPython
+        # 3.5.4 on Windows. Without this, occurrence probability of the issue
+        # increases to approx. 30 ~ 50 % from 0 % (0 out of 240).
+        stdout.write("")
```

Comment:
In some contexts, `sys.stdout` can be None or not writable, so this should either be checked for, exception caught, or another approach taken. Does the write itself have to execute, or is it just the reference to stdout, e.g. `getattr(stdout, 'write')`?

Reply:
Thank you. Exceptional cases (None or not writable) were totally out of my consideration. OK, let me try & seek a better option like you suggested.

Reply:
It's super wired but I found that `locals()` is sufficient. Because I suppose the `locals()` would be always safe, that's the best option I have ever imagined. If you have some idea, please let me know and I will try to make it one step ahead once again.

Reply:
Weeeird...But if it works, should be safe. Thanks for digging in!